### ISSUE TITLE
Elixir GenStage crashes when receiving demand from Ergo

### ISSUE THREAD

**Issue by okkdev** [ergo-services/ergo#103]
**Describe the bug**
Elixir GenStage crashes when receiving demand from Ergo.
Error: 
```
[error] GenServer Producer terminating
** (FunctionClauseError) no function clause matching in GenStage.maybe_producer_cancel/2
```

**To Reproduce**
I made a small repo with the reproduction steps:
https://github.com/okkdev/ergo_elixir_genstage_issue

**Expected behavior**
Have the Elixir Producer handle the demand and Ergo do the work.

**Environment (please complete the following information):**
 - Arch: arm64
 - OS: macOS
 - Framework Version v1.999.210
 - Number of CPU: 10

**Comment by halturin** [comment]
Thanks for the detailed report. I'll take a look

**Comment by halturin** [comment]
fixed in 2.1.1
